huaweicloud/services/servicestage/resource_huaweicloud_servicestage_repository_authorization.go

```diff
@@ -0,0 +1,174 @@
+					validation.StringLenBetween(4, 63),
+				),
+			},
+			"type": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"devcloud", "bitbucket", "github", "gitlab", "gitee",
+				}, false),
+			},
+			"host": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				RequiredWith: []string{"token"},
+				ExactlyOneOf: []string{"user_name"},
+			},
+			"token": {
+				Type:      schema.TypeString,
+				Optional:  true,
+				ForceNew:  true,
+				Sensitive: true,
+			},
+			"user_name": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				RequiredWith: []string{"password"},
+			},
+			"password": {
+				Type:      schema.TypeString,
+				Optional:  true,
+				ForceNew:  true,
+				Sensitive: true,
+			},
+		},
+	}
+}
+
+func resourceRepositoryAuthCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var err error
+	config := meta.(*config.Config)
+	client, err := config.ServiceStageV1Client(config.GetRegion(d))
+	if err != nil {
+		return fmtp.DiagErrorf("error creating ServiceStage v1 client: %s", err)
+	}
+
+	rType := d.Get("type").(string)
+	switch rType {
+	case "github", "gitlab", "gitee":
+		host, ok := d.GetOk("host")
+		if !ok {
+			return diag.Errorf("the host must be set if the type is github, gitlab or gitee.")
+		}
+
+		opt := repositories.PersonalAuthOpts{
+			Name:  d.Get("name").(string),
+			Host:  host.(string),
+			Token: d.Get("token").(string),
+		}
+		auth, err := repositories.CreatePersonalAuth(client, rType, opt)
+		if err != nil {
+			return diag.Errorf("error creating the ServiceStage token authorization: %s", err)
+		}
+
+		d.SetId(auth.Name)
+	case "devcloud", "bitbucket":
+		userName, ok := d.GetOk("user_name")
+		if !ok {
+			return diag.Errorf("the user_name must be set if the type is devcloud or bitbucket.")
+		}
+
+		s := fmt.Sprintf("%s:%s", userName.(string), d.Get("password").(string))
+		opt := repositories.PwdAuthOpts{
+			Name:  d.Get("name").(string),
+			Token: base64.StdEncoding.EncodeToString([]byte(s)),
+		}
+		auth, err := repositories.CreatePwdAuth(client, rType, opt)
+		if err != nil {
+			return diag.Errorf("error creating the ServiceStage password authorization: %s", err)
+		}
+
+		d.SetId(auth.Name)
+	default:
+		return diag.Errorf("unsupported authorization type: %s", rType)
+	}
+
+	return resourceRepositoryAuthRead(ctx, d, meta)
+}
+
+func resourceRepositoryAuthRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	region := config.GetRegion(d)
+	client, err := config.ServiceStageV1Client(region)
+	if err != nil {
+		return fmtp.DiagErrorf("error creating ServiceStage v1 client: %s", err)
+	}
+
+	resp, err := repositories.List(client)
+	if err == nil {
```

Comment:
suggest to use 'if err != nil ' to make the code style same with other resources

Reply:
the current logical structure is easier to understand with less code.